Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/ProgressIndicators/ProgressIndicatorObjectDisplay.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Threading.Tasks;
+using TMPro;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Extensions.SceneTransitions
+{
+    /// <summary>
+    /// This class manages how a gameobject rotates and/or scales
+    /// when activated as part of a Progress Indicator effect.
+    /// </summary>
+    public class ProgressIndicatorObjectDisplay : MonoBehaviour, IProgressIndicator
+    {
+        public Transform MainTransform { get { return transform; } }
+        public ProgressIndicatorState State { get { return state; } }
+        public float Progress { set { progress = value; } }
+        public string Message { set { messageText.text = value; } }
+
+        [SerializeField]
+        private Transform scaleTargetObject = null;
+        [SerializeField]
+        private Transform rotateTargetObject = null;
+
+        [Header("How fast does object rotate?")]
+        [SerializeField]
+        private float rotationIncrement = 200;
+
+        [Header("Start scale of the object?")]
+        [SerializeField]
+        private float minScale = 1.0f;
+
+        [Header("Final scale of the object?")]
+        [SerializeField]
+        private float maxScale = 9.0f;
+
+        [Header("Should object rotate?")]
+        [SerializeField]
+        private bool rotationActive = false;
+
+        [Header("Rotation occurs about which axes?")]
+        [SerializeField]
+        private bool xAxisRotation = false;
+        [SerializeField]
+        private bool yAxisRotation = true;
+        [SerializeField]
+        private bool zAxisRotation = false;
+
+        [Header("Grow / Shrink curve on open / close")]
+        [SerializeField]
+        [Tooltip("Curve for opening animation. MUST end with value >= 1 or transition will not complete.")]
+        private AnimationCurve openCurve = AnimationCurve.EaseInOut(0, 0, 1, 1);
```

Comment:
Nit: Is there any way to validate that these things are correct (i.e. in OpenAsync or CloseAsync, could they Debug.Warn or something if they're being misused?)

Reply:
Otherwise the infinite loop is gonna be a bad time (and possibly hard to debug for clients)